src/storage/checkpoint_manager.cpp

```diff
@@ -169,157 +179,112 @@ void SingleFileCheckpointReader::LoadFromStorage() {
 	con.Commit();
 }
 
+void CheckpointWriter::WriteEntry(CatalogEntry &entry, Serializer &serializer) {
+	serializer.WriteProperty(99, "catalog_type", entry.type);
```

Comment:
@Maxxen currently this is written in a set order, so there is no need to write a CatalogType for every entry, but with this PR everything can be interleaved so we write an additional "catalog_type" for every entry